vllm/worker/habana_model_runner.py

```diff
@@ -193,6 +193,12 @@ def forward(self, *args, **kwargs):
         hidden_states = self.model(*args, **kwargs)
         hidden_states = hidden_states.view(-1, hidden_states.shape[-1])
         hidden_states = hidden_states.index_select(0, selected_token_indices)
+        from vllm.lora.layers import VocabParallelEmbeddingWithLoRA
+        if isinstance(self.model.model.embed_tokens,
```

Comment:
@hlahkar please correct this, right now this check is tied to Llama2, we need to make this generic so it works across models.
Also add a comment on why we need this piece of code.

Reply:
This change is now updated to be model agnostic